qiskit/transpiler/_transpiler.py

```diff
@@ -48,7 +47,7 @@ def transpile(circuits, backend, basis_gates=None, coupling_map=None, initial_la
     Raises:
         TranspilerError: in case of bad compile options, e.g. the hpc options.
     """
-    if isinstance(circuits, QuantumCircuit):
+    if not isinstance(circuits, list):
```

Comment:
This really changes the semantics. I would prefer:
```suggestion
    from qiskit import _quantumcircuit 
    if isinstance(circuits, _quantumcircuit.QuantumCircuit):
```